src/behave_tests/features/environment.py

```diff
@@ -0,0 +1,99 @@
+
+import os
+
+from behave import fixture
+from jinja2 import Template
+
+KCLI_PLANS_DIR = "kcli_plans"
+
+KCLI_PLAN_TEMPLATE = """
```

Comment:
Here we have a small problem... what uses kcli as plan is a "jinja template" , and we are processing the template and providing a plan with all the replacements needed. That could work .. but is not enough flexible

We need to produce a jinja template for the kcli plan: i mean, we need to produce something completely identical  to:
https://github.com/jmolmo/kcli-plans/blob/ceph_dev/ceph/ceph_cluster.yml

Look, we have in the right "ceph_cluster.yml" file  things that are going to be processed by kcli in order to produce a plan, with the current implementation you are doing part of the kcli work, and that is not correct. 
For example, you are generating directly the configuration for each of the nodes... we really want kcli to do this part of the job.

Reply:
@jmolmo I have made the changes.